src/Workspaces/Core/Portable/Shared/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -190,9 +187,41 @@ private static bool IsAbstractPropertyImplemented(INamedTypeSymbol classOrStruct
                 interfacesOrAbstractClasses,
                 IsImplemented,
                 ImplementationExists,
-                GetMembers,
+                (INamedTypeSymbol type, ISymbol within) =>
+                {
+                    if (type.TypeKind == TypeKind.Interface)
+                    {
+                        return type.GetMembers().WhereAsArray(m => m.DeclaredAccessibility == Accessibility.Public &&
+                                                                   m.Kind != SymbolKind.NamedType && IsImplementable(m) &&
+                                                                   !isPropertyWithNonPublicImplementableAccessor(m));
+                    }
+
+                    return type.GetMembers();
+                },
                 allowReimplementation: false,
                 cancellationToken: cancellationToken);
+
+            static bool isPropertyWithNonPublicImplementableAccessor(ISymbol m)
```

Comment:
1. same comments as above
2. once made into a local function, can we give better names for parameters.

thanks!